### PR TITLE
Add support for Float, Int64, UInt64

### PR DESCRIPTION
**Pull request by abarisain**
Hello,

Thanks for your work on the library! It has been very useful to easily add expectations in my Swift project.

I have noticed that my functions explicitly handling Int64 values were failing when I used numbers too big for an Int.

```swift
func testEqual_int64s() {
        let ret = self.verifier.equal(Int64.max, to: Int64.max)
        XCTAssertTrue(ret)
}
```

After investigating, it turns out that when a value doesn't fit into an Int, the library can't compare them.
I tweaked Int+MockUsable to handle all FixedWidthIntegers, and then explicitly added support for UInt64 and Int64. I kept Int and UInt, as it's better not to force the library to use 64bit words if not necessary, even if performance isn't that important in this kind of testing library.

Further testing showed that Floats didn't work out of the box either, so I also added support for that.

Side note: It was quite hard to debug, maybe the library could use some sort of debug mode to understand why an argument doesn't match (the debugger reported Int64 on both the expectation and actual value, which was confusion).

### CODE REVIEW

**Comment by abarisain** [comment]
Sorry for the commit mess!

There should be a way to reduce the boilerplate in Int+MockUsable.swift by introducing a "AnyProviding" protocol, and implement it like this:

```
public protocol AnyProviding {
    static var any: Self { get }
}

extension MockUsable where Self: AnyProviding {
    public static var anyValue: MockUsable {
        return Self.any
    }
}
```

I did not do that, as the reduced boilerplate was not worth the complexity and new public API.

**Comment by abarisain** [comment]
I'm not sure I get the codecov output here. I compared coverage locally and Xcode shows no changes.

```
...tMockTests/Mocks/ArgumentVerifyMandatoryMock.swift | 40.00% <0.00%> (-10.00%)
-- | --
```

I don't see any change in this file's coverage using Xcode's coverage sidebar.

File: Sources/InstantMock/MockUsable/Float+MockUsable.swift
Comment (pirishd) [inline Sources/InstantMock/MockUsable/Float+MockUsable.swift]: Hey @abarisain, you can put your name in the header file 😉

File: Tests/InstantMockTests/Unit Tests/MockUsable/FloatMockUsableTests.swift
Comment (pirishd) [inline Tests/InstantMockTests/Unit Tests/MockUsable/FloatMockUsableTests.swift]: Same here, you can put your name 😉

**Review by pirishd — CHANGES_REQUESTED** [review]
Thanks a lot @abarisain for this great contribution 😀!

Would you please update the pull request:
- with the two tiny updates in the file headers of the added files
- update the `README.md` file, paragraph [MockUsable > Supported Type](https://github.com/pirishd/InstantMock#supported-types), to add the new types now supported?

Once done, I will release a new version right away 😉

**Comment by abarisain** [comment]
Thank you for the really fast answer! I'll make the changes as soon as I can get to it.

Maybe hold off on a release, I have another PR coming up adding Date support: I added it locally in my tests, but I think that everyone can benefit from it!

File: Sources/InstantMock/MockUsable/Float+MockUsable.swift
Comment (abarisain) [reply Sources/InstantMock/MockUsable/Float+MockUsable.swift]: Done! I'm leaving your copyright line though, to be consistent with the project. Thanks :)

**Comment by abarisain** [comment]
PR has been updated. Totally missed the readme, thank you for catching that.

I'll submit the Date PR once this has been merged (I'd like to avoid a conflict on the readme)

**Review by pirishd — APPROVED** [review]

**Pull request merged**
